src/database/DataTypes/RequisitionItem.js

```diff
@@ -294,6 +285,7 @@ RequisitionItem.schema = {
     id: 'string',
     requisition: { type: 'Requisition', optional: true },
     item: { type: 'Item', optional: true },
+    itemName: 'string',
```

Comment:
The `item` is `optional` so maybe in some edge case there's no `item` to get the `itemName`?  Does it need to be `optional` also? Also the existing lines?

Reply:
> 
> 
> Great solution! Couple minors.
> 
> I also think we need to consider migration. Essentially just need to iterate over all requisition items (which aren't finalised, I guess) and assign the related items name to `itemName`.
> 
> A little concerned about the migration as there are sites with 10's of 1000's of requisitions..

That's a very good point. I'm assuming the majority of these are finalised (?), if so, this might be mitigated by filtering on finalised (since Realm is lazy, this should avoid pulling in any finalised records). Does Realm for RN provide any decorators for indexing fields etc?

Reply:
You can add `indexed` into the schema declaration object

Reply:
Follow up: under what circumstances are the items undefined? Would this only be due to a partial sync? In that case, would it be better to do something like below?

```
itemName: { type: 'string', default: 'placeholder' },
```

Reply:
I guess some sync thing for the item being undefined. I'm unsure, it's just theoretically possible (so 100% going to happen!).